sort_by_nil_fix - fix sort_by nil

7sedam7 opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/kontena/kontena?pullRequest=2542) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/kontena/kontena?pullRequest=2542) before we can accept your contribution.<br/><hr/>**Joel Mislov Kunst** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/kontena/kontena?pullRequest=2542) it.</sub>

SpComb commented:
Hmm, what situation does this happen in? It is actually possible for the `node_number` to be `nil`, but that should only happen for a very short period of time while a node is connecting.

Are you perhaps running into something like the race condition in #2071?

Just a little surprised that this popped up now, right after restoring the original `sort_by` in 1.3.2.

7sedam7 commented:
Signed CLA.

@SpComb  I think it's not #2071, I do not know how I got to it, but I was not able to get rid of it, so I just changed it for my self and added PR in case it's ok for you as well.

SpComb commented:
Your git commit author doesn't match up with your GitHub account's email addresses, so the cla-assistant.io check didn't pass yet... you need to either add that email address to your github profile and re-check, or rewrite the commit author and force-push?

SpComb commented on cli/lib/kontena/cli/nodes/list_command.rb:
```diff
@@ -70,7 +70,7 @@ def node_data
         end
 
-        grid_nodes.sort_by { |n| n['node_number'] }
+        grid_nodes.sort_by { |n| n['node_number'].to_i }
```
Could also have a spec case with `{ 'node_number' => nil }` (see `cli/spec/kontena/cli/nodes/list_command_spec.rb`)

SpComb commented:
@7sedam7 this is still pending on the CLA check passing, because your git commit author does not match up with your github email address.

SpComb commented:
The server should now be fixed in #2694 to always assign a `node_number` to nodes, so this shouldn't be a problem on the CLI anymore... I hope.
